mongo/datadog_checks/mongo/collectors/custom_queries.py

```diff
@@ -128,6 +130,9 @@ def _collect_custom_metrics_for_query(self, client, raw_query):
             for metric_name, metric_value, submit_method in metric_info:
                 submit_method(metric_name, metric_value, tags=query_tags)
 
+        if empty_result_set:
+            raise Exception('Custom query returned an empty result set.')
```

Comment:
why an exception and not a log.error/warning?

Reply:
That function is ran per-query, and the check runs each query in it's own try/catch.
So the exception makes sense in the context of this method because the method is not able to do anything. But that won't fail the whole check, nor custom queries collection as the exception is caught and logged by the check.